core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java

```diff
@@ -99,7 +100,7 @@ private MapEntryMapper(RowMapper<K> keyMapper, RowMapper<V> valueMapper) {
 
     @Override
     public Map.Entry<K, V> map(ResultSet rs, StatementContext ctx) throws SQLException {
-        return immutableEntry(keyMapper.map(rs, ctx), valueMapper.map(rs, ctx));
+        return ImmutableEntry.of(keyMapper.map(rs, ctx), valueMapper.map(rs, ctx));
```

Comment:
You could consider using `AbstractMap.SimpleEntry` which might remove the need for the `ImmutableEntry` class.